cmd/listen-notification-handlers.go

```diff
@@ -141,6 +141,13 @@ func (api objectAPIHandlers) ListenNotificationHandler(w http.ResponseWriter, r
 		peer.Listen(listenCh, ctx.Done(), values)
 	}
 
+	var emptyEventTicker <-chan time.Time
+	if values.Get("ping") == "true" {
+		ticker := time.NewTicker(time.Minute)
+		defer ticker.Stop()
+		emptyEventTicker = ticker.C
+	}
+
```

Comment:
You can change keepAlive to send empty events{} entirely @vadmeste

Reply:
> You can change keepAlive to send empty events{} entirely @vadmeste

This is possible if we require users to upgrade their minio-go version when they upgrade MinIO version, otherwise it will break

Reply:
No what I meant is never send empty instead send an event called s3:Ping event - clients know exactly what to look for.

No new changes are needed on client side.

Reply:
A new flag just for this is an overkill IMO.

Reply:
@harshavardhana s3:ping may break some deployments, like if an application is listening only for object upload events it will receive other events and will break.

> A new flag just for this is an overkill IMO.

I don't think we have a choice unless to fix minio-go as well. For example, avoid using bufio.Scanner or implement a reader which discards whitespaces between JSON documents (not very obvious since it needs to ignore escaped '{' and '}' as well)

At the other hand, we can take this opportunity to deprecate the whitespace code and send s3:Ping instead. The client can send a request with '?ping=60' to ask the server to send s3:Ping each 60 seconds instead of  the whitespaces.